### PR TITLE
Sort all mounts by destination directory

### PR DESCRIPTION
**Pull request by rhatdan**
Currently depending on the sort order of mount points, we can overmount
a volume specified from the user. Podman has a function sortMount that
sorts all mounts based on destination directory to ensure all mounts
show up. This PR moves the function from Podman to Buildah. Once merged
I will change Podman to use the buildah function.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rhatdan** [comment]
Replaces https://github.com/containers/buildah/pull/3114

File: util/util.go [hunk 1]
```diff
@@ -474,3 +476,26 @@ func MergeEnv(defaults, overrides []string) []string {
...
+}
+
+func (m byDestination) parts(i int) int {
+	return strings.Count(filepath.Clean(m[i].Destination), string(os.PathSeparator))
+}
+
+func SortMounts(m []specs.Mount) []specs.Mount {
+	sort.Sort(byDestination(m))
+	return m
+}
```
Comment (nalind) [inline util/util.go:501]: This needs unit tests.

**Comment by gabemontero** [comment]
fwiw I was able to pull the util.go and run_linux.go changes from here into an openshift/builder development image, run it against an OCP user provisioned install on top of rhel worker nodes, and verify that host entitlements work

we have a work around in openshift/builder for this, but ideally we'd like to revert that and pull in this change into openshift/builder's vendor tree through normal go mod vendor processing.

thanks

**Comment by gabemontero** [comment]
whether we can wait for this getting into something openshift/builder can vendor in vs. move forward with our workaround (which results in some noisy warning messages when we run on RHCOS) most likely will depend on whether any of growing number of customer use cases around host entitlements for OCP builds results in some escalations or not

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by gabemontero** [comment]
any chance @TomSweeneyRedHat 's LGTM can be translated into an lgtm label?  Or are there additional reviews you all are waiting on?

the OCP build customer cases around host entitlements continues to grow, and ideally we'd like our official fix to include this change vendored into https://github.com/openshift/builder instead of the "openshift/builder-only" workaround we current have.

@bparees FYI

**Comment by TomSweeneyRedHat** [comment]
@gabemontero we need another LGTM before it can merge.  @rhatdan @nalind @vrothberg all happy green test buttons, can you slap on a LGTM and get this merged please?

**Comment by rhatdan** [comment]
I see @gabemontero comments as a LGTM

**Pull request merged**